### ISSUE TITLE
About pretrain and finetune

### ISSUE THREAD

**Issue by TNTWEN** [mx54039q/entroformer#4]
你好！
之前我将entroformer在compressai框架下重构，加载了您提供的预训练模型，成功训练出了您论文的效果。最近尝试从头开始训练模型，发现从头开始训练有些困难，所以和您确认一下超参数的设置与训练方案的改进。

数据集： Openimage搜集的1.5W张图片       训练的模型均为parallel自回归

Pretrain:
根据https://github.com/damo-cv/entroformer/blob/main/scripts/pretrain.sh
我在Pretrain训练时，batshsize 为8，epoch设置为1000，λ 0.3，学习率全程保持1e-4 ，patchSize 256 。采用了random mask，mask_ratio为 0.5。

Finetune：
batshsize 为8，取消了random mask，patchSize 384 ，λ 0.02 ，学习率初始化为1e-4，学习率变化策略尝试了您提供的策略，其中epoch 500，也尝试了factor=0.5,patience=20的ReduceLROnPlateau学习率调整策略，这两者差别不大。但训练结果比使用预训练模型差了些。

在Kodak测试集上，加载预训练模型结果为 	**Loss：1.017  bpp：0.607 PSNR：35.275。**  
但是pretrain+finetune的结果就差了好些：**Loss：1.036  bpp：0.6181 PSNR：35.2041。** 

特别是模型从学习率将为2.5e-05开始，Loss就减少的很小。

故有以下几点超参数的设定想和您确认，希望您能提供一些改进训练的思路：
①：Pretrain时您将patchSize设置为 256，而非384。由于epoch多， patchSize设256时虽然训练地更快些，但是否会影响pretrain模型的效果？
②：您的batchsize都是设置为8，该训练任务是否对batchsize敏感，更大的batchsize是否会有帮助？
③：我的训练结果给我的感觉是陷入了局部最优点，在Finetune时，前25个epoch为warmup，75个epoch学习率是1e-4，感觉训得有些少，是否可以finetune时学习率保持1e-4，训练更多的epoch来寻找全局最优点？

**Comment by mx54039q** [comment]
1. pretrain的patchsize设为384效果是可能会好一些的, 但是训练时间会大大增加.
2. batchsize设为8 效果更好, 图像压缩任务不是batchsize越大越好; 而且模型中没有使用bn, 所以也没有使用大batchsize的必要性.
3. 是可以的.

是否可以po一下你pretrain模型的结果 (bpp \ PSNR), 特别是psnr, 如果没有达到42.6的话, 说明main autoencoder没有训好. 因为pretrain阶段如果没训好, 反倒是会使得模型陷入局部最优, 导致finetune时无法进一步优化.

**Comment by TNTWEN** [comment]
@mx54039q 
这是我pretrain最后的结果： λ 0.3    ，测试集是Kodak
Test epoch 999: Average losses:3.124 |	MSE loss: 0.00005 |	PSNR : 42.7741 |	Bpp loss: 2.0720

**Comment by mx54039q** [comment]
pretrain的结果没有问题. 你可以尝试一下finetune的时候也保持lr1e-4的学习率, 将step lr去掉.

**Comment by guoguo1314** [comment]
你们好啊，我有个疑问就是预训练的时候Test epoch 999: Average losses:3.124 | MSE loss: 0.00005 | PSNR : 42.7741 | Bpp loss: 2.0720，测出来结果psnr是35.275，这是一种正常现象吗？验证集和测试集怎么差这么多。
训练ms-ssim-loss的时候出现这个问题，训练mse-loss是没问题的，都是使用的一样的训练集和训练策略以及一样的各种epoch，学习率，batch_size等。我还在怀疑是否模型对ssim这个损失函数过拟合。

**Comment by mx54039q** [comment]
我已收到你的邮件。                                               by    钱一琛
